lib/dead_end/core_ext.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+# Allow lazy loading, only load code if/when there's a syntax error
+autoload :DeadEnd, "dead_end/api"
```

Comment:
I doubt if this works after `DeadEnd` is defined in `ruby_opt_init`.
`autoload` does nothing for already defined constant.

Reply:
Good to know. I will try to switch to a different constant.

Reply:
Here is my new approach/proposal https://github.com/zombocom/dead_end/pull/148